### ISSUE TITLE
Build crashing on Ventura 13.0 (M1 Pro)

### ISSUE THREAD

**Issue by dramikei** [timkurvers/valheim-macos#4]
Successfully created a build of Valheim by following the guide however, upon launching valheim.app the game crashes after displaying valheim logo for a second.

Attaching the crash report below:

https://pastebin.com/e9hBSviF

**Comment by timkurvers** [comment]
That's unfortunate, thanks for attaching the crash report.

It seems to be crashing when establishing a connection to Steam. Could you double check that Steam is running and you are logged in with an account that owns Valheim?

**Comment by dramikei** [comment]
> That's unfortunate, thanks for attaching the crash report.
> 
> It seems to be crashing when establishing a connection to Steam. Could you double check that Steam is running and you are logged in with an account that owns Valheim?

Yes, my steam is running and own Valheim. I even tried exiting steam and trying again. still the same issue :(

**Comment by timkurvers** [comment]
Could you have a look at `Library/Logs/IronGate/Valheim/Player.log` in your user's home folder? That may have some additional indicators on why it's crashing.

(If you'd like to share the file here, censor out any personal information beforehand)

**Comment by dramikei** [comment]
Heres the log:
https://pastebin.com/Sra62jEF

**Comment by zenangst** [comment]
@dramikei the first thing that I would try is to try running it from a different location. `Desktop` and `Documents` gets a bit special treatment in macOS because it is inside a cloud container if you have iCloud syncing enabled.

**Comment by dramikei** [comment]
I dont have icloud sync on, tried running it from Documents as well still the same result :(

**Comment by timkurvers** [comment]
Thanks for giving that a shot 🙌 

There's a few other things @zenangst and I discussed looking at the crash report:

- System Integrity Protections seems to be disabled. If you do not absolutely need it, you could try turning SIP back on again and see if that changes behaviour.

- macOS Ventura 13.0 is not the latest point-version, and more recent versions may have fixed Rosetta 2 bugs that you may be running into.

If you'd rather not alter your system too much, one last test could be to make a development build. To do so, change `nondevelopment` to `development` in `build.sh`:

```diff
diff --git a/build.sh b/build.sh
index 008f120..f973c42 100755
--- a/build.sh
+++ b/build.sh
@@ -12,7 +12,7 @@ version="0.214.2"
 # Unity configuration
 unityversion="2020.3.33f1"
 unityhash="915a7af8b0d5"
-variant="macos_x64_nondevelopment_mono"
+variant="macos_x64_development_mono"
 
 confirm() {
   read -r -p "${1:-Are you sure?} [y/N] " response
```

Then run `./build.sh` again and test the build. 🙏

**Comment by zenangst** [comment]
Yeah I would just put it in your home directory, because as I said before, both of those folders do require special permissions:

Take this as an example:

<img width="452" alt="image" src="https://user-images.githubusercontent.com/57446/228169211-1fe20a9a-63b7-4e27-abe8-17e9f40d49ef.png">

Here, the Arc browser has to ask specifically for permissions to access those folders properly. However, this might just be a red-harring in terms of what is going wrong. But it feels like a good thing to just verify that it doesn't cause any issues.

**Comment by davak** [comment]
Same problems here.  It was working fine before the latest update to the game.  I decided to update so I could continue to play on the server with my friends. 

I did a git pull and repeated the entire process.  Since then, I get a crash after the game title screen.

**Comment by davak** [comment]
I even deleted everything and started over from scratch.  Same error.  I wonder if something about the latest Valheim update broke your process.

**Comment by timkurvers** [comment]
> I even deleted everything and started over from scratch. Same error. I wonder if something about the latest Valheim update broke your process.

Ugh, that's unfortunate. Curious why this patch in particular caused problems. 🤔

Are you seeing anything of use in the `Library/Logs/IronGate/Valheim/Player.log` file in your home folder?

**Comment by dramikei** [comment]
@timkurvers I got it working by pulling latest changes using `git pull` and creating a development build. Non-development build however still seems to crash 🤔 
SIP off, and Ventura 13.0.

**Comment by dramikei** [comment]
Also one more thing,

> Steam overlay is non-functional. For an alternative FPS counter, press F2 in-game.

we can actually get steam overlay to work by adding Valheim.app as a non steam game :D

**Comment by rquast** [comment]
Not sure if it helps, but it works on Ventura 13.1 M1 Pro 16" for me, but it takes ages to initially load a game (5 minutes in some cases) and it will crash out if steam isn't loaded in the background when you get past the title screen. Only thing for me that that doesn't work properly is cloud sync.

**Comment by timkurvers** [comment]
> Not sure if it helps, but it works on Ventura 13.1 M1 Pro 16" for me, but it takes ages to initially load a game (5 minutes in some cases) and it will crash out if steam isn't loaded in the background when you get past the title screen. Only thing for me that that doesn't work properly is cloud sync. Running from my home directory.

Does it take five minutes even when joining an existing world? If so, that sounds slightly lengthy indeed.

With regards to cloud sync, there's [an open issue](https://github.com/timkurvers/valheim-macos/issues/10) where we're trying to find out what causes these issues. Just in case I made these changes after you made your build: check out the README for some more info on how to investigate cloud syncing issues.

**Comment by rquast** [comment]
> Does it take five minutes even when joining an existing world? If so, that sounds slightly lengthy indeed.
> 
> With regards to cloud sync, there's [an open issue](https://github.com/timkurvers/valheim-macos/issues/10) where we're trying to find out what causes these issues. Just in case I made these changes after you made your build: check out the README for some more info on how to investigate cloud syncing issues.

The five minutes issue I later found out was caused by the game files not syncing properly, so it was generating a new world for existing games (hence the long wait). I manually copy the games at the moment.. have played for about 10 hours without issue. Have also upgraded to Ventura 13.3.1 without issue too.

**Comment by timkurvers** [comment]
Closing this issue, feel free to comment/re-open if problems still exist.
